bundles/binding/org.openhab.binding.sallegra/src/main/java/org/openhab/binding/sallegra/internal/SallegraGenericBindingProvider.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sallegra.internal;
+
+import org.openhab.binding.sallegra.SallegraBindingProvider;
+import org.openhab.core.items.Item;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is for parsing the items configuration.
+ * 
+ * @author Benjamin Marty (Developed on behalf of Satelco.ch)
+ * @since 1.8.0
+ */
+public class SallegraGenericBindingProvider extends AbstractGenericBindingProvider implements SallegraBindingProvider {
+
+	
+	private static Logger logger = LoggerFactory
+			.getLogger(SallegraGenericBindingProvider.class);
+	
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "sallegra";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * 
+	 * [deviceId:command:value]
+	 * 
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item,
+			String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
```

Comment:
move this to the end because otherwise it is being added to internal data structures, while later in the method you may throw a BindingConfigParseException.